CorsixTH/Lua/map.lua

```diff
@@ -422,7 +426,7 @@ function Map:loadMapConfig(filename, config, custom)
         -- Overlay with the specific configuration for this level
         errors, config = self:loadMapConfig(difficulty .. level_number .. ".SAM", config)
         if errors then return "Overlay level number must be 1-12. Current value is " .. level_number end
-        -- Finally overlay additional CorsixTH config per level
+        -- Finally overlay additional CorsixTH config per level. As these files may not exist, failure is permitted.
         local level_path = self.app:getFullPath({"Levels", "original" .. level_number .. ".level"})
         _, config = self:loadMapConfig(level_path, config, true)
```

Comment:
Not your fault, but blindly assuming that a load works is a bad idea.

The loaded data gets used I assume, which means that all kinds of things can go wrong if the loaded data is not correct or the load fails. As it's not caught here, it will cause trouble elsewhere at a spot that has nothing to do with the cause of the error. It will take quite some time before it is tracked back to here.

It's fine to assume the load does not fail often enough to spend time on error reporting. However do check the result. The simplest is an `assert`.
If that triggers, you saved yourself a week of debugging. If it happens more frequently, something more convenient should probably done, but that's a few steps ahead currently.

As you didn't introduce this, I am fine with not fixing it here.